pages/software/python.md

```diff
@@ -0,0 +1,457 @@
+# Python Options
+
+There are many options for using Python and its derivates on CCR's systems.  CCR provides, at most, two versions of Python and two versions of Anaconda as well as SciPy (Scientific Python libraries) via it's modules system.  These are updated once per year with the oldest version being rotated out.  Users will see a warning when a module is being deprecated giving you advance notice to update your workflow.  
+
+**This page describes the different options and we recommend you read through it all to determine which is best for your research needs**
+
+## Recommended Organization for Python  
+
+For these examples, we will assume you have access to a group shared project directory located in `/projects/academic` or `/projects/industry`  If you do not already have a directory for yourself within your group's project directory, create one.  Then we recommend creating a directory structure for your software and Python environments.  We recommend, and use throughout this documentation, this structure where `yourgroup=your group's directory name` and `$USER=your username`  
+
+```
+ccruser@vortex ~ $ mkdir /projects/academic/yourgroup/$USER  
+ccruser@vortex ~ $ mkdir /projects/academic/yourgroup/$USER/software
+ccruser@vortex ~ $ mkdir /projects/academic/yourgroup/$USER/software/python  
+ccruser@vortex ~ $ mkdir /projects/academic/yourgroup/$USER/software/python/venvs
+ccruser@vortex ~ $ mkdir /projects/academic/yourgroup/$USER/software/anaconda
+ccruser@vortex ~ $ mkdir /projects/academic/yourgroup/$USER/software/anaconda/envs
+```
+
+!!! Note  
+    Not all users have access to a shared project directory.  Please verify your access via [ColdFront](../portals/coldfront.md).  
+
+## Using CCR's Python Installation  
+
+CCR provides as part of the currently supported toolchains.  We do not recommend using a version included within the operating system nor the one provided in CCR's compatibility layer.  These can differ across login and compute nodes and can be replaced with newer versions, potentially affecting your project.  These are also compiled to make the best use out of the architecture you're running on rather than offering one solution for the lowest common denominator.  Rather, using one contained in CCR's software repository toolchain provides you the most flexibility and usability across all CCR systems.  We highly recommend using python virtual environments for your projects.  A python virtual environment can be created using a specific version of python and can be used in isolation of other environments.  This is convenient if your projects require different versions of Python packages or have differing dependencies.  Within the virtual environment you can install additional Python packages  These virtual environments can be stored in your home directory or within your group's project directory making them available to other members of your group.  
```

Comment:
It's perfectly fine to use the python from the compat layer and in our new base os images there will be no python. So we can simplify this section quite a bit, maybe something like: 

> CCR provides as python part of our software modules.  We highly recommend using python virtual environments for your projects.  A python virtual environment can be created using a specific version of python and can be used in isolation of other environments.  This is convenient if your projects require different versions of Python packages or have differing dependencies.  Within the virtual environment you can install additional Python packages  These virtual environments can be stored in your home directory or within your group's project directory making them available to other members of your group.

Reply:
fixed